### DISCUSSION TITLE
Github setup for Dapp Deploy

### DISCUSSION THREAD

**Discussion by Hofstud** [hashlips-lab/nft-erc721-collection#357] [General]
Hi @liarco,

Firstly I would like to thank you for all of your amazing dev work and tips in helping me on the Hashlips NFT creation journey. Its been incredible seeing how creative people can be with yours and Daniels guidance.

My query is how to setup the Github repository for the first time to prepare for cloud-flare from your video ?

This part isnt very clear to me. I would appreciate your help to clarify this section a bit more :)

Once again thanks heaps!
cheers,
Leonie

**Comment by Hofstud** [comment]
Its okay - Ive figured it out - the 1st remote setup of Github went ok for the main net - just took a while to update and push the files. As I havent done this procedure before I didnt want to break anything

**Reply by locache** [reply]
Hello, I am having the same issue. How did you setup the Github repository for the first time to prepare for cloud-flare.  

I have opened the entire nft-erc721 folder in Visual Studio. When I click on the Source Control Tab on the left I see "Initialize Repository." If I click this button then I have 60 files that need to be committed but I don't believe this is correct.  What am I supposed to do here? Which folder am I supposed to run the 'new repository commands in ?" Thank you!

**Comment by locache** [comment]
Hey @Hofstud Did you see my comment above? How did you fix your issue? When I deploy the DAPP to Cloudflare I get the following errors: 

" Installing missing commands
11:25:56.913    Verify run directory
11:25:56.913    Executing user command: yarn build-dapp
11:25:56.917    /opt/build/bin/build: line 39: yarn: command not found
11:25:56.923    Failed: build command exited with code: 127
11:25:57.680    Failed: an internal error occurred"

**Reply by Hofstud** [reply]
Hi @locache,

I was a bit confused about setting up the github repository as it was m first time doing this step.

So what I did was:
Copy/paste the lines into visual code terminal from GitHub when you setup a NEW repository. Its the same bunch of code lines that you see in liarco's video.

I then selected the commit and sync to GitHub via visual studio. It takes a little while for the first time, then you will see in your new GitHub repository all the files that are needed for cloudflare.
Only when the GitHub files are setup will Cloudflare work. It took about 2-3 goes to refresh cloudflare to work but once both github and cloudflare are synced, you should be okay !

**Reply by Hofstud** [reply]
On Liarco's video https://www.youtube.com/watch?v=uUrbIXUgVz4&t=354s at 1.33 sec, this is the code that I copy/paste that is created when u setup a new githhub repository into visual studio.
Hope this helps
 
<img width="1132" alt="Screen Shot 2022-08-05 at 8 40 07 am" src="https://user-images.githubusercontent.com/96272109/182978173-b03517ca-10fc-4d58-8e4e-96e1d94b7705.png">

**Reply by locache** [reply]
@Hofstud I really appreciate you trying to help me out...I mean it! I just tried that again but I keep getting the same error on CloudFlare.

I'm using the "erc721 2.4.2" file. I ran the NEW commands in the Nft-erc721-collection-2.4.4 root folder (61 commits): 

![image](https://user-images.githubusercontent.com/32690966/182982002-99297bcf-3da0-4b3f-be87-fe7c4afa5f05.png)

Committed and pushed the files to github:

![image](https://user-images.githubusercontent.com/32690966/182982097-93d2ed2f-50d0-46e8-9dba-30a3530b00e5.png)

Synced with Cloudflare then ran the deployment and still get the same error:
 
21:21:01.945 | Cloning repository...
-- | --
21:21:03.206 | From https://github.com/locache/spf3
21:21:03.206 | * branch            88945905f8ab7f66ffad2c6205b4cf3e810061f0 -> FETCH_HEAD
21:21:03.206 |  
21:21:03.279 | HEAD is now at 8894590 my changes
21:21:03.279 |  
21:21:03.422 |  
21:21:03.448 | Success: Finished cloning repository files
21:21:04.136 | Installing dependencies
21:21:04.147 | Python version set to 2.7
21:21:07.183 | v12.18.0 is already installed.
21:21:08.298 | Now using node v12.18.0 (npm v6.14.4)
21:21:08.500 | Started restoring cached build plugins
21:21:08.514 | Finished restoring cached build plugins
21:21:08.952 | Attempting ruby version 2.7.1, read from environment
21:21:12.279 | Using ruby version 2.7.1
21:21:12.630 | Using PHP version 5.6
21:21:12.779 | 5.2 is already installed.
21:21:12.805 | Using Swift version 5.2
21:21:12.805 | Started restoring cached node modules
21:21:12.819 | Finished restoring cached node modules
21:21:13.051 | Installing NPM modules using NPM version 6.14.4
21:21:14.046 | npm notice created a lockfile as package-lock.json. You should commit this file.
21:21:14.127 | up to date in 0.166s
21:21:14.128 | found 0 vulnerabilities
21:21:14.128 |  
21:21:14.137 | NPM modules installed
21:21:14.454 | Installing Hugo 0.54.0
21:21:15.128 | Hugo Static Site Generator v0.54.0-B1A82C61A/extended linux/amd64 BuildDate: 2019-02-01T10:04:38Z
21:21:15.131 | Started restoring cached go cache
21:21:15.148 | Finished restoring cached go cache
21:21:15.278 | go version go1.14.4 linux/amd64
21:21:15.297 | go version go1.14.4 linux/amd64
21:21:15.299 | Installing missing commands
21:21:15.300 | Verify run directory
21:21:15.300 | Executing user command: yarn build-dapp
21:21:15.304 | /opt/build/bin/build: line 39: yarn: command not found
21:21:15.310 | Failed: build command exited with code: 127
21:21:16.102 | Failed: an internal error occurred

Cloudflare can't understand the yarn build-dapp command for some reason?? (Let me know if you cant see those screenshots and attachments)

**Reply by LGW22** [reply]
> @Hofstud I really appreciate you trying to help me out...I mean it! I just tried that again but I keep getting the same error on CloudFlare.
> 
> I'm using the "erc721 2.4.2" file. I ran the NEW commands in the Nft-erc721-collection-2.4.4 root folder (61 commits):
> 
> ![image](https://user-images.githubusercontent.com/32690966/182982002-99297bcf-3da0-4b3f-be87-fe7c4afa5f05.png)
> 
> Committed and pushed the files to github:
> 
> ![image](https://user-images.githubusercontent.com/32690966/182982097-93d2ed2f-50d0-46e8-9dba-30a3530b00e5.png)
> 
> Synced with Cloudflare then ran the deployment and still get the same error:
> 
> 21:21:01.945	Cloning repository...
> 21:21:03.206	From https://github.com/locache/spf3
> 21:21:03.206	* branch 88945905f8ab7f66ffad2c6205b4cf3e810061f0 -> FETCH_HEAD
> 21:21:03.206	 
> 21:21:03.279	HEAD is now at 8894590 my changes
> 21:21:03.279	 
> 21:21:03.422	 
> 21:21:03.448	Success: Finished cloning repository files
> 21:21:04.136	Installing dependencies
> 21:21:04.147	Python version set to 2.7
> 21:21:07.183	v12.18.0 is already installed.
> 21:21:08.298	Now using node v12.18.0 (npm v6.14.4)
> 21:21:08.500	Started restoring cached build plugins
> 21:21:08.514	Finished restoring cached build plugins
> 21:21:08.952	Attempting ruby version 2.7.1, read from environment
> 21:21:12.279	Using ruby version 2.7.1
> 21:21:12.630	Using PHP version 5.6
> 21:21:12.779	5.2 is already installed.
> 21:21:12.805	Using Swift version 5.2
> 21:21:12.805	Started restoring cached node modules
> 21:21:12.819	Finished restoring cached node modules
> 21:21:13.051	Installing NPM modules using NPM version 6.14.4
> 21:21:14.046	npm notice created a lockfile as package-lock.json. You should commit this file.
> 21:21:14.127	up to date in 0.166s
> 21:21:14.128	found 0 vulnerabilities
> 21:21:14.128	 
> 21:21:14.137	NPM modules installed
> 21:21:14.454	Installing Hugo 0.54.0
> 21:21:15.128	Hugo Static Site Generator v0.54.0-B1A82C61A/extended linux/amd64 BuildDate: 2019-02-01T10:04:38Z
> 21:21:15.131	Started restoring cached go cache
> 21:21:15.148	Finished restoring cached go cache
> 21:21:15.278	go version go1.14.4 linux/amd64
> 21:21:15.297	go version go1.14.4 linux/amd64
> 21:21:15.299	Installing missing commands
> 21:21:15.300	Verify run directory
> 21:21:15.300	Executing user command: yarn build-dapp
> 21:21:15.304	/opt/build/bin/build: line 39: yarn: command not found
> 21:21:15.310	Failed: build command exited with code: 127
> 21:21:16.102	Failed: an internal error occurred
> Cloudflare can't understand the yarn build-dapp command for some reason?? (Let me know if you cant see those screenshots and attachments)

hi there, do you commit all 61 file? got error? because this few day i keep try and get error too. than group people call me copy that "git init....." paste it on terminal ,but remove the readme.md. than it success and upload the source to github. my one look not same as your, i dont have github file, minting dapp, smart contract file. is because i did something wrong?

**Comment by Hofstud** [comment]
Hey Locache,



You are doing all the commands that I did so I dont know why its not working for you. I could see the images.



This is my github after running the commands and committing the files through visual studio and this is the minting dapp after deploy



https://github.com/Hofstud/HOFS-Minting_Dapp





https://hofs-minting-dapp.pages.dev/





If you can escalate up to Liarco somehow - try asking in hashlips discord...



 








---- On Fri, 05 Aug 2022 09:30:12 +0800 locache ***@***.***> wrote ---





https://github.com/Hofstud I really appreciate you trying to help me out...I mean it! I just tried that again but I keep getting the same error on CloudFlare.

I'm using the "erc721 2.4.2" file. I ran the NEW commands in the Nft-erc721-collection-2.4.4 root folder (61 commits):

https://user-images.githubusercontent.com/32690966/182982002-99297bcf-3da0-4b3f-be87-fe7c4afa5f05.png

Committed and pushed the files to github:

https://user-images.githubusercontent.com/32690966/182982097-93d2ed2f-50d0-46e8-9dba-30a3530b00e5.png

Synced with Cloudflare then ran the deployment and still get the same error:

https://github.com/hashlips-lab/nft-erc721-collection/files/9264487/cloudflare.log.log

It can't understand the yarn build-dapp command for some reason?? (Let me know if you cant see those screenshots and attachments)
